crowdin/src/main/java/com/crowdin/platform/data/DataManager.kt

```diff
@@ -272,27 +272,32 @@ internal class DataManager(
 
     @WorkerThread
     fun getTicket(event: String): String? {
-        var ticketValue: String? = null
-        val type = object : TypeToken<MutableMap<String, TicketItem>>() {}.type
-        var ticketsMap: MutableMap<String, TicketItem>? = localRepository.getData(EVENT_TICKETS, type)
-        if (ticketsMap == null) {
-            ticketsMap = mutableMapOf()
-        }
+        try {
+            var ticketValue: String? = null
+            val type = object : TypeToken<MutableMap<String, TicketItem>>() {}.type
+            var ticketsMap: MutableMap<String, TicketItem>? = localRepository.getData(EVENT_TICKETS, type)
+            if (ticketsMap == null) {
+                ticketsMap = mutableMapOf()
+            }
 
-        val ticketItem = ticketsMap[event]
-        if (ticketItem == null || ticketItem.isExpired()) {
-            Log.d(CROWDIN_TAG, "Ticket expired for event: $event")
-            remoteRepository.getTicket(event)?.data?.ticket?.let {
-                ticketsMap[event] = TicketItem(it, System.currentTimeMillis() + EVENT_TICKETS_EXPIRATION)
-                localRepository.saveData(EVENT_TICKETS, ticketsMap)
-                ticketValue = it
+            val ticketItem = ticketsMap[event]
+            if (ticketItem == null || ticketItem.isExpired()) {
+                Log.d(CROWDIN_TAG, "Ticket expired for event: $event")
+                remoteRepository.getTicket(event)?.data?.ticket?.let {
+                    ticketsMap[event] = TicketItem(it, System.currentTimeMillis() + EVENT_TICKETS_EXPIRATION)
+                    localRepository.saveData(EVENT_TICKETS, ticketsMap)
+                    ticketValue = it
+                }
+            } else {
+                Log.d(CROWDIN_TAG, "Ticket not expired for event: $event")
+                ticketValue = ticketItem.ticket
             }
-        } else {
-            Log.d(CROWDIN_TAG, "Ticket not expired for event: $event")
-            ticketValue = ticketItem.ticket
-        }
 
-        return ticketValue
+            return ticketValue
+        } catch (throwable: Throwable) {
+            Log.e(CROWDIN_TAG, "Ticket failed", throwable)
+            return ""
```

Comment:
@FacundoAramayo, thanks for your contribution!

Here, return `null` instead of `""`. Since we use the `dataManager.getTicket(suggestionEvent)?.let` function, this will prevent an unnecessary subscribe event.

Everything else looks good to me.